modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -85,6 +86,14 @@ public ServiceTemplateEntity updateServiceTemplate(String id, Ocl ocl) {
             log.error(errMsg);
             throw new ServiceTemplateNotRegistered(errMsg);
         }
+        CurrentUserInfo currentUserInfo = identityProviderManager.getCurrentUserInfo();
+        if (Objects.nonNull(currentUserInfo)) {
+            if (!StringUtils.equals(existedService.getNamespace(),
+                    currentUserInfo.getNamespace())) {
+                throw new AccessDeniedException("No right to update service template belong to "
```

Comment:
No permissions to update services templates belonging to other namespaces.

Reply:
Done.